exercises/circular-buffer/example.js

```diff
@@ -1,25 +1,28 @@
 let buffer, bufferMax;
 
-const BufferEmptyException = () => ({
-    name: "buffer empty exception!",
-    message: "can't read from an empty buffer!"
-  });
-
-const BufferFullException = () => ({
-    name: "buffer full exception!",
-    message: "can't write to a full buffer!"
-  });
+export class BufferEmptyException extends Error {
+  constructor(message) {
+    super();
+    this.message = message || 'Buffer is empty.';
+  }
+};
+export class BufferFullException extends Error {
+  constructor(message) {
+    super();
+    this.message = message || 'Buffer is full.';
+  }
```

Comment:
The constructor part is optional. But it reads nicely in the stacktrace with the message.